docs/data/date-pickers/custom-field/PickerWithJoyField.tsx

```diff
@@ -302,14 +336,31 @@ interface JoyDateFieldProps
     > {}
 
 function JoyDateField(props: JoyDateFieldProps) {
-  const { inputRef: externalInputRef, slots, slotProps, ...textFieldProps } = props;
+  const { inputRef: externalInputRef, ...textFieldProps } = props;
 
-  const response = useDateField<Dayjs, typeof textFieldProps>({
+  const { onClear, clearable, slots, slotProps, ...fieldProps } = useDateField<
+    Dayjs,
+    typeof textFieldProps
+  >({
     props: textFieldProps,
     inputRef: externalInputRef,
   });
+  const { InputProps: ProcessedInputProps, fieldProps: processedFieldProps } =
+    useClearField<
+      {},
```

Comment:
This feels hacky to me, but I'm having trouble wrapping my head around why `typeof textFieldProps` is clashing with the types in `useClearField` so I'm open to suggestions 😄

Reply:
I meant that maybe we could at least mention, that this section is optional and can be omitted if the clearable behavior is not needed, but it's a small nitpick. 😉